core/src/main/java/com/linecorp/armeria/client/retry/RetryConfig.java

```diff
@@ -0,0 +1,205 @@
+            long responseTimeoutMillisForEachAttempt) {
+        checkArguments(maxTotalAttempts, responseTimeoutMillisForEachAttempt);
+        this.maxContentLength = maxContentLength;
+        this.retryRuleWithContent = requireNonNull(retryRuleWithContent, "retryRuleWithContent");
+        fromRetryRuleWithContent = RetryRuleUtil.fromRetryRuleWithContent(retryRuleWithContent);
+        this.maxTotalAttempts = maxTotalAttempts;
+        this.responseTimeoutMillisForEachAttempt = responseTimeoutMillisForEachAttempt;
+        needsContentInRule = true;
+        retryRule = null;
+    }
+
+    private static void checkArguments(int maxTotalAttempts, long responseTimeoutMillisForEachAttempt) {
+        checkArgument(
+                maxTotalAttempts > 0,
+                "maxTotalAttempts: %s (expected: > 0)",
+                maxTotalAttempts);
+        checkArgument(
+                responseTimeoutMillisForEachAttempt >= 0,
+                "responseTimeoutMillisForEachAttempt: %s (expected: >= 0)",
+                responseTimeoutMillisForEachAttempt);
+    }
+
+    /**
+     * Converts this {@link RetryConfig} to a mutable {@link RetryConfigBuilder}.
+     */
+    public RetryConfigBuilder<T> toBuilder() {
+        final RetryConfigBuilder<T> builder =
+                needsContentInRule ?
+                builder(retryRuleWithContent).maxContentLength(maxContentLength) : builder(retryRule);
+        return builder
+                .maxTotalAttempts(maxTotalAttempts)
+                .responseTimeoutMillisForEachAttempt(responseTimeoutMillisForEachAttempt);
+    }
+
+    /**
+     * Returns the maximum allowed number of total attempts made by a {@link RetryingClient}.
+     */
+    public int maxTotalAttempts() {
+        return maxTotalAttempts;
+    }
+
+    /**
+     * Returns the response timeout for each attempt in milliseconds.
+     * When requests in {@link RetryingClient} are made,
+     * corresponding responses are timed out by this value.
+     */
+    public long responseTimeoutMillisForEachAttempt() {
+        return responseTimeoutMillisForEachAttempt;
+    }
+
+    /**
+     * Returns the {@link RetryRule} used by {@link RetryingClient} using this config, could be null.
+     */
+    @Nullable
+    public RetryRule retryRule() {
+        return retryRule;
+    }
+
+    /**
+     * Returns the {@link RetryRuleWithContent} used by {@link RetryingClient} using this config, could be null.
+     */
+    @Nullable
+    public RetryRuleWithContent<T> retryRuleWithContent() {
+        return retryRuleWithContent;
+    }
+
+    /**
+     * Returns config's {@code maxContentLength}, which is non-zero only if
+     * a {@link RetryRuleWithContent} is used.
+     */
+    public int maxContentLength() {
+        return maxContentLength;
+    }
+
+    /**
+     * Returns whether a {@link RetryRuleWithContent} is being used.
+     */
+    public boolean needsContentInRule() {
+        return needsContentInRule;
```

Comment:
Could be `retryRuleWithContent != null`?

Reply:
Done